src/dmd/expressionsem.d

```diff
@@ -313,9 +313,33 @@ Expression resolveOpDollar(Scope* sc, ArrayExp ae, IntervalExp ie, Expression* p
         return e;
     }
 
+    int is_dollar = 0;
+
+    auto lwr = ie.lwr;
+    auto upr = ie.upr;
+
+    // check if lower or upper were dollar
+    if ((cast(IdentifierExp)lwr))
```

Comment:
This does not work as expected for `extern(C++)` classes IIRC
```suggestion
    if (wr.isIdentifierExp())
```

Ditto for the other dynamic casts